Fixing mpas-ocean gpu compile

dazlich opened the pull request:
Mpas-ocean needs its own variable for gpu compilation flags, adding the -gpu=redundant option. This option causes the gpu and cpu solutions to match.

Commit:
Create cupid workflow

Created template.cupid file and added a workflow to run it. Added module
load conda to run successfully for. Added a load conda command to the
config_machines.xml file for derecho. Updated the scratch path on
casper.

Commit:
Get rid of code that loads conda and add code that will ensure the commands in the template are run with the correct environment set up. Add module load nco to config_machines.xml file

jedwards4b committed:
update nvhpc and gnu on derecho

jedwards4b committed:
update derecho port

mnlevy1981 committed:
Merge tag 'ccs_config_cesm1.0.6' into add_cupid_workflow

ccs_config_cesm1.0.6

mnlevy1981 committed:
Move generate_cupid_config_file into CUPiD

template.cupid assumes CUPiD is in the CESM source tree (tools/CUPiD) and runs
the generate script from there. Also cleaned up some white space

mnlevy1981 committed:
Update template.cupid call to generate config

generate_cupid_config_file.py now expects some commandline arguments, so the
cupid template as been modified accordingly.

mnlevy1981 committed:
Changed API for CUPiD script

sjsprecious committed:
update derecho setups for new GPU XML variables

jedwards4b committed:
Merge pull request #189 from sjsprecious/update_gpu_config

Update Derecho setups for new GPU XML variables

jedwards4b committed:
add memory usage variables for use on derecho

jedwards4b committed:
remove anonymous ftp and gridftp, use wget

jedwards4b committed:
wget needs the trailing /

jedwards4b committed:
fix for izumi nag build

jedwards4b committed:
Merge pull request #193 from jedwards4b/nag_netcdf_fix_izumi

fix for izumi nag build

jedwards4b committed:
Merge pull request #192 from jedwards4b/update_inputdata_server

remove anonymous ftp and gridftp, use wget

jedwards4b committed:
update default ww3 grid to mg37

jedwards4b committed:
update grid aliases for wg37

jedwards4b committed:
reduce optimazation for fms2_io methods

jedwards4b committed:
turn off checks with ifx

mnlevy1981 committed:
Merge tag 'ccs_config_cesm1.0.8' into add_cupid_workflow

ccs_config_cesm1.0.8

jedwards4b committed:
update Depends.nvhpc

jedwards4b committed:
Merge pull request #190 from jedwards4b/add_mem_variables

add memory usage variables for use on derecho

jedwards4b committed:
back out intel compiler change

jedwards4b committed:
seperate intel and intel-oneapi cmake macros files

jedwards4b committed:
exceptions for oneapi compiler

jedwards4b committed:
Merge pull request #195 from jedwards4b/update_default_ww3grid

Changes for compsets, grids and compilers on derecho

jedwards4b committed:
add a cam and a clm file to debug exceptions for nvhpc

jedwards4b committed:
Merge branch 'main' into derecho_updates_080224

jedwards4b committed:
Merge pull request #179 from jedwards4b/derecho_updates_080224

update nvhpc and gnu on derecho

jedwards4b committed:
Merge pull request #196 from jedwards4b/nvhpc_more_debug_issues

add a cam and a clm file to debug exceptions for nvhpc

jedwards4b committed:
backout derecho module update

jedwards4b committed:
Merge pull request #197 from jedwards4b/backout_derecho_updates

backout derecho module update

mnlevy1981 committed:
Updates for CUPiD API changes

mnlevy1981 committed:
Add a few local variables to template.cupid

These will eventually be populated from an env_?.xml file, but for now starting
a list of variables we want the user to be able to set.

Also added two minor updates:
1. unset PYTHONPATH to ensure we are only using conda-installed python packages
2. The API for the script to generate the ADF config file has been changed - we
now point to the directory containing a CUPiD config file rather than the CUPiD
example

jedwards4b committed:
update to ncarenv2412

dazlich committed:
Update 15km file defaults

The default ocean/seaice files for 15km have been updated to
use a new one that does not initially generate unstable transients.

These modifications are in the mpas-ocean, mpas-seaice, and
ccs_config components.

mnlevy1981 committed:
Call cupid scripts directly

Don't assume that (cupid-infrastructure) is pointing to the right CUPiD version

Commit:
Merge branch '15km/ocean' into 'ew-develop' (PR #33)

Update input files for MPAS-O and MPAS-SI at 15km resolution.

alperaltuntas committed:
fix the default grid name for wav components

jedwards4b committed:
Merge pull request #200 from ESMCI/fix_ww3_default_grid

fix the default grid name for wav components

gustavo-marques committed:
Update JRA to tx2_3v2 Mapping File

Updated the mapping file to use a 250 km e-folding scale and radius
of spreading.

jedwards4b committed:
Merge pull request #176 from ingridc2051/add_cupid_workflow

Create cupid workflow

alperaltuntas committed:
Merge pull request #201 from gustavo-marques/update_map_jra_to_tx2_3_runoff

Update JRA to tx2_3v2 Mapping File

gustavo-marques committed:
Fix typo, sm ~> nnsm

alperaltuntas committed:
Merge pull request #203 from gustavo-marques/update_map_jra_to_tx2_3_runoff

Fix typo, sm ~> nnsm

sjsprecious committed:
add new CLUBB code for GPU compilation

sjsprecious committed:
add back some changes

sjsprecious committed:
change the place

wwieder committed:
Update modelgrid_aliases_nuopc.xml

mask information was  needed to run an I-case with the f09_t232 grid.  Is this true for other grids too?

fischer-ncar committed:
Merge pull request #205 from wwieder/patch-1

Enable t232 mask on default land grid

jedwards4b committed:
Merge pull request #204 from sjsprecious/add_new_clubb_code

Add new CLUBB code for GPU compilation

TeaganKing committed:
very preliminary updates in template.cupid for cupid workflow

TeaganKing committed:
Update template.cupid with bash += string format

TeaganKing committed:
Update template.cupid for glc

brian-eaton committed:
add ne30pg3_ne30pg3_t232

brian-eaton committed:
change ne30pg3_ne30pg3_t232 to ne30pg3_ne30pg3_mt232

TeaganKing committed:
Update template.cupid in response to Bill's review

TeaganKing committed:
Update template.cupid to move if postprocessing

TeaganKing committed:
Update template.cupid - logic fix for all components flag

TeaganKing committed:
Update template.cupid to include new arguments for generate_cupid_config_for_cesm_case.py

TeaganKing committed:
Update template.cupid to use cupid example directory rather than PWD

mnlevy1981 committed:
Updates to CUPiD workflow

Teagan and I added a bunch of xml variables to env_postprocessing.xml (in a
CUPiD PR), so template.cupid now uses those variables to modify the CUPiD
config. Also, we moved case.cupid into the default workflow (but
RUN_POSTPROCESSING is FALSE, so it won't run unless the user changes that
variable to TRUE)

TeaganKing committed:
Update template.cupid to include ADF xml var

TeaganKing committed:
Update template.cupid in response to Bill's review

dazlich committed:
Update hires file information

An entry is created for the regionally refined 60-3km
grid with the refinement over the western U.S. The
3.75 and 7.5km hires meshes are updated to new, stable
initial conditions.

jedwards4b committed:
change gnu debug option from Og to O0 g

mnlevy1981 committed:
Merge branch 'derecho_update_ncarenv2412' into cupid_workflow_updates

mnlevy1981 committed:
Move batch directives above function definition

Also cleaned up formatting of environment variables and added a new argument to
generate_cupid_config_for_cesm_case.py (to make sure ADF output ends up in the
correct place)

jedwards4b committed:
Merge pull request #211 from brian-eaton/cam_t232

add ne30pg3_ne30pg3_t232

jedwards4b committed:
Merge pull request #210 from TeaganKing/cupid_workflow_updates

Updates in template.cupid and config_workflow.xml for cupid workflow

mnlevy1981 committed:
Renamed some XML variables

CUPID now uses same STARTDATE format that rest of CIME uses

mnlevy1981 committed:
Updated names of CUPID variables

Also, cupid wants inputs in date format, rather than integers now

mnlevy1981 committed:
Change command line argument names in CUPiD

need to reflect new names when calling the script

mnlevy1981 committed:
case.cupid calls script owned by CUPiD

Adding a new script to CUPiD will make it easier to modify the workflow as more
diagnostic packages are added

jedwards4b committed:
updates to esmf, netcdf and parallelio

jedwards4b committed:
Merge pull request #213 from jedwards4b/derecho_update_ncarenv2412

update to ncarenv2412

jedwards4b committed:
add toronto backup for inputdata

jedwards4b committed:
back off pio update because esmf requires older version

jedwards4b committed:
back off pio/netcdf update because esmf requires older version

jedwards4b committed:
Merge pull request #216 from jedwards4b/toronto_inputdata

add toronto backup for inputdata

jedwards4b committed:
Merge pull request #217 from jedwards4b/derecho_update_revised

back off pio update because esmf requires older version

jedwards4b committed:
Merge branch 'main' into more_cupid_workflow_updates

jedwards4b committed:
Merge pull request #214 from mnlevy1981/more_cupid_workflow_updates

More cupid workflow updates

jedwards4b committed:
remove esmf/8.8.0-debug due to a spack issue

jedwards4b committed:
Merge pull request #218 from jedwards4b/derecho_update_revised

remove esmf/8.8.0-debug due to a spack issue

jedwards4b committed:
back off intel compiler to 2024.2.1

jedwards4b committed:
reapply intel/2025 and add a Depends for file mo_rxt_rates_conv.F90

jedwards4b committed:
make the -O change only for newer gnu compilers

sjsprecious committed:
update the Casper configuration

sjsprecious committed:
update the mpi command

jedwards4b committed:
finally looking good

jedwards4b committed:
Merge pull request #219 from jedwards4b/back_out_ifx

back off intel compiler to 2024.2.1

sjsprecious committed:
Merge branch 'main' into update_casper_config

sjsprecious committed:
use gpu_type attribute

sjsprecious committed:
use Derecho's combination to pass DEBUG test

jedwards4b committed:
Merge pull request #220 from sjsprecious/update_casper_config

Update machine configuration for Casper

brian-eaton committed:
add F compset grids w/ mt232 mask

brian-eaton committed:
add mpasa480_mpasa480_mt232

alperaltuntas committed:
specify not_compset attributes for non-MOM6 ocn grids

jedwards4b committed:
revert the update to gcc compiler due to cice issue

jedwards4b committed:
fix esmf module load

jedwards4b committed:
Merge pull request #223 from jedwards4b/backout_gcc_update

Backout gcc update

jedwards4b committed:
fix a cism build issue by excluding ldflags from cmake

Commit:
Merge tag ccs_config_cesm1.0.8 from ESMCI/ccs_config_cesm into 'ew-develop'

Update EarthWorksOrg/ccs_config_cesm with upstream work from
ESCOMP/CESM/cesm3_0_beta04

fischer-ncar committed:
Merge pull request #224 from jedwards4b/fix/cism_bld_issue

fix a cism build issue by excluding ldflags from cmake

Commit:
Merge tag ccs_config_cesm1.0.19 from ESMCI/ccs_config_cesm into 'ew-develop'

Update EarthWorksOrg/ccs_config_cesm with upstream work from
ESCOMP/CESM/cesm3_0_beta05

sjsprecious committed:
update derecho and casper cmake configs for kokkos build

sjsprecious committed:
add missing cmake files

sjsprecious committed:
update kokkos setups and machine modules

sjsprecious committed:
update lib for lapack and blas

sjsprecious committed:
load math lib

sjsprecious committed:
update modules

sjsprecious committed:
bug fix for casper machine

sjsprecious committed:
remove lapack and blas on Casper

jedwards4b committed:
Merge pull request #222 from alperaltuntas/mom_not_compset

Specify not_compset attribute for non-MOM6 resolutions

jedwards4b committed:
Merge pull request #221 from brian-eaton/mt232_grids

Add F compset grids w/ mt232 mask

sjsprecious committed:
remove gpu_type attr for some machine features

sjsprecious committed:
add missing CPP variable for nvhpc

sjsprecious committed:
load kokkos on Derecho for intel compiler by default

sjsprecious committed:
remove EKAT dependency

sjsprecious committed:
remove openmp for nvhpc compiler

jedwards4b committed:
Merge pull request #225 from sjsprecious/build_kokkos_in_cam

Build kokkos in the CAM workflow and clean up the NCAR machine configuration files

Commit:
Merge branch 'update/cesm3_0_beta05/ccs_config_cesm1.0.19' into 'ew-develop' (PR #35)

jtruesdal committed:
cmake file updates for izumi nh

jtruesdal committed:
mods for intel derecho

alperaltuntas committed:
update wg37 mesh to cdf5

jedwards4b committed:
Merge pull request #227 from alperaltuntas/wg37_mesh_update

Update wg37 mesh to cdf5

sjsprecious committed:
clean up kokkos setups

sjsprecious committed:
remove kokkos flags

sjsprecious committed:
add missing kokkos flags

sjsprecious committed:
add restrict flag for C/C++ code

sjsprecious committed:
update kokkos build for serial target

sjsprecious committed:
clean up more

sjsprecious committed:
update casper machine and batch configurations

jedwards4b committed:
Merge pull request #229 from sjsprecious/update_casper_setup

Update Casper machine and batch configurations

jtruesdal committed:
updates to build theta-l/kokkos with Intel, NAG, and GCC on izumi

jtruesdal committed:
izumi cmake updates

Commit:
Merge branch 'update/ocean3p75' into 'ew-develop' (PR #36)

Bring in updates for higher resolution input data, MPAS-O GPU updates,
fixes for diagnostic values in restarts for MPAS-O and MPAS-SI, and
using a different time manager for MPAS-O and MPAS-SI.

brian-eaton committed:
fix aliases for C48_C48_mg17, C96_C96_mg17, C192_C192_mg17

sjsprecious committed:
more cleanup for cmake options

sjsprecious committed:
Revert "izumi cmake updates"

This reverts commit fb3619d8515b2ea6caf3778beadc37cafafd7cf1.

sjsprecious committed:
Revert "updates to build theta-l/kokkos with Intel, NAG, and GCC on izumi"

This reverts commit be35df8e0777c9054b3a4597c6c96be1f51e4f4b.

sjsprecious committed:
Revert "cmake file updates for izumi nh"

This reverts commit bf415e0814abb27339fed41a74e2678ab228d280.

sjsprecious committed:
remove redundant GNU Fortran flags

jedwards4b committed:
Merge pull request #231 from sjsprecious/clean_up_kokkos

Clean up Kokkos setups

jedwards4b committed:
Merge pull request #230 from brian-eaton/fix_C_grids

fix aliases for C48_C48_mg17, C96_C96_mg17, C192_C192_mg17

sjsprecious committed:
enable cuda target correctly on Derecho

sjsprecious committed:
add missing Hommexx_enable_gpu definition

jedwards4b committed:
Merge pull request #232 from sjsprecious/enable_kokkos_cuda_target

Enable the CUDA target of Kokkos library

Commit:
Merge tag ccs_config_cesm1.0.40 from ESMCI/ccs_config_cesm into 'ew-develop' (PR #37)

Update EarthWorksOrg/ccs_config_cesm with upstream work from
ESCOMP/CESM:cesm3_0_beta06

Commit:
Revert to nvhpc/24.3 stack on Derecho

Continue to use older modules due to EarthWorksOrg/EarthWorks Issue#90 -
memory leak with stacks using nvhpc/24.7 or newer.

Commit:
Add -DCPRPGI back, prevent build issues with CDEPS

Without this flag a 'call destroy(Sdoc)' statement in
dshr_stream_mod.F90 causes compilation to fail with nvhpc/24.3

Commit:
erge branch 'postBeta06_restore_nv243' into ew-develop (PR #38)

Continue to work around memory issue with software stacks using nvhpc
newer than v24.4 on Derecho.

Commit:
Merge branch 'ew-develop' into 'ew-main' for v2.5 release.

gdicker1 commented:
@dazlich, it looks like I didn't get everything pushed correctly last week as we seem to be missing the release-ccs_config-ew2.5 and ccs_config-ew2.5.000 tags I'd expect and ew-main and ew-develop branches aren't at the commits they should be.

Can this wait for a few days for NCAR's GLADE Scratch to come back online from maintenance? Then I can get the tags pushed from the sandbox I was working in. Apologies for the issues that causes this PR and other work.

If needed, I could sort this out more quickly.

dazlich commented:
No hurry necessary for my part. Better to be orderly. I am an mpas mentor at DCMIPthis week so I won’t be doing much EarthWorks.

gdicker1 commented:
Understood, I'll keep an eye out.

Good luck with DCMIP!

gdicker1 commented:
@dazlich, could you recreate this PR with the GPU changes you need?

I think something was off since I didn't push EWOrg/ccs_configs branches and tags a couple of weeks ago. This PR shouldn't have counted as merged, but I also didn't see any GPU changes that you would have added.

dazlich commented:
will do

On Jun 10, 2025, at 11:40 AM, G. Dylan Dickerson ***@***.***> wrote:


** Caution: EXTERNAL Sender **

[https://avatars.githubusercontent.com/u/43178166?s=20&v=4]gdicker1 left a comment (EarthWorksOrg/ccs_config_cesm#40)<https://github.com/EarthWorksOrg/ccs_config_cesm/pull/40#issuecomment-2960111267>

@dazlich<https://github.com/dazlich>, could you recreate this PR with the GPU changes you need?

I think something was off since I didn't push EWOrg/ccs_configs branches and tags a couple of weeks ago. This PR shouldn't have counted as merged, but I also didn't see any GPU changes that you would have added.

—
Reply to this email directly, view it on GitHub<https://github.com/EarthWorksOrg/ccs_config_cesm/pull/40#issuecomment-2960111267>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ADS3XWEBW4KXMDPONJXSAYD3C4KB7AVCNFSM6AAAAAB6N6VBA2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDSNRQGEYTCMRWG4>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

dazlich commented:
Ok, strange

I am looking at ccs_config_cesm/machines/cmake_macros/nvhpc.cmake in GitHub for the fix-ocean-gpu branch.
The code I added has the string ‘MPASO_OPENACC_GPU_FLAGS’ in it. There are three lines I added with it.

It turns out I put one of them in the wrong place so I am fixing that and committing that.


On Jun 10, 2025, at 11:40 AM, G. Dylan Dickerson ***@***.***> wrote:


** Caution: EXTERNAL Sender **

[https://avatars.githubusercontent.com/u/43178166?s=20&v=4]gdicker1 left a comment (EarthWorksOrg/ccs_config_cesm#40)<https://github.com/EarthWorksOrg/ccs_config_cesm/pull/40#issuecomment-2960111267>

@dazlich<https://github.com/dazlich>, could you recreate this PR with the GPU changes you need?

I think something was off since I didn't push EWOrg/ccs_configs branches and tags a couple of weeks ago. This PR shouldn't have counted as merged, but I also didn't see any GPU changes that you would have added.

—
Reply to this email directly, view it on GitHub<https://github.com/EarthWorksOrg/ccs_config_cesm/pull/40#issuecomment-2960111267>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ADS3XWEBW4KXMDPONJXSAYD3C4KB7AVCNFSM6AAAAAB6N6VBA2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDSNRQGEYTCMRWG4>.
You are receiving this because you were mentioned.Message ID: ***@***.***>
